java/src/main/kotlin/io/spine/validation/java/generate/option/BoundedFieldGenerator.kt

```diff
@@ -0,0 +1,210 @@
+                var violation = ${violation(ReadVar("fieldPath"), ReadVar("typeName"), value)};
+                $violations.add(violation);
+            }    
+            """.trimIndent()
+        )
+    }
+
+    /**
+     * The number type of the bound.
+     *
+     * It is checked for unsigned primitives, for which the generator reports
+     * a compilation warning. Such primitives are not natively supported in Java.
+     */
+    protected abstract val boundPrimitive: NumericBound.ValueCase
+
+    /**
+     * Returns a boolean expression that checks if the given [value] is within the bounds.
+     *
+     * Note that unsigned values must be handled in a special way because Java does not
+     * support them natively. For this, `java.lang.Integer` and `java.lang.Long` classes
+     * provide static methods to treat signed types as unsigned, including parsing, printing,
+     * comparison and math operations.
+     */
+    protected abstract fun isOutOfBounds(value: Expression<Number>): Expression<Boolean>
+
+    private fun violation(
+        fieldPath: Expression<FieldPath>,
+        typeName: Expression<TypeName>,
+        fieldValue: Expression<Number>,
+    ): Expression<ConstraintViolation> {
+        val qualifiedName = field.qualifiedName
+        val typeNameStr = typeName.stringify()
+        val placeholders = supportedPlaceholders(fieldPath, typeNameStr, fieldValue)
+        val errorMessage = templateString(view.errorMessage, placeholders, option, qualifiedName)
+        return constraintViolation(errorMessage, typeNameStr, fieldPath, fieldValue)
+    }
+
+    protected abstract fun supportedPlaceholders(
+        fieldPath: Expression<FieldPath>,
+        typeName: Expression<String>,
+        fieldValue: Expression<*>,
+    ): Map<ErrorPlaceholder, Expression<String>>
+
+    /**
+     * Returns a string representation of this [NumericBound].
+     *
+     * Note that `int` and `long` values that represent unsigned primitives are printed as is.
+     * In the rendered Java code, they can become negative number constants due to overflow,
+     * which is expected.
+     */
+    protected fun NumericBound.asLiteral() =
+        when (valueCase) {
+            FLOAT_VALUE -> Literal("${floatValue}F")
+            DOUBLE_VALUE -> Literal("$doubleValue")
+            INT32_VALUE -> Literal("$int32Value")
+            INT64_VALUE -> Literal("${int64Value}L")
+            UINT32_VALUE -> Literal("$uint32Value")
+            UINT64_VALUE -> Literal("$uint64Value")
+            else -> error(
+                "Unexpected field type `$valueCase` when converting range bounds to Java literal." +
+                        " Make sure the policy, which verified `${view::class.simpleName}`," +
+                        " correctly filtered out unsupported field types."
+            )
+        }
+}
+
+private fun unsignedIntegerWarning(file: File, span: Span) =
+    Compilation.warning(file, span) {
```

Comment:
Reiterating the comment which I [made for already closed PR](https://github.com/SpineEventEngine/validation/pull/200/files#r2012570908):

> [!NOTE]
> We can probably store a number of times we issue this warning and put the full text only once. On second and after occasions we just print a short text referring to the same issue having some text sounding like "Already reported".
>
> Gradle does a similar thing for repeated errors.
>
>  This way the console log won't be "polluted".
>

Reply:
In other words, let's show this long warning text only once for the first field with the unsigned bounds set for a signed type. Then, have a shorter text that mentions the field name, its type and the fact that the full warning was reported before.

@armiol, FYI.

Reply:
This comment applies for the `(range)` option too. We can probably combine the warnings or have their "branches" depending of the option kind.

Reply:
We can probably have qualifiers for warnings similarly to those that ErrorProne and detekt have. Some titles with `CamelNotation`. 

This will allow:

1. Referencing the full version of text from shorter ones.
2. Suppressing them during the compilation process via ProtoData options.